gallery/gallery/lib/demos/cupertino/cupertino_refresh_demo.dart

```diff
@@ -0,0 +1,83 @@
+// Copyright 2019 The Flutter team. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:math' show Random;
+
+import 'package:flutter/cupertino.dart';
+import 'package:flutter/material.dart';
+import 'package:gallery/l10n/gallery_localizations.dart';
+
+// BEGIN cupertinoRefreshDemo
+
+class CupertinoRefreshControlDemo extends StatefulWidget {
+  @override
+  _CupertinoRefreshControlDemoState createState() =>
+      _CupertinoRefreshControlDemoState();
+}
+
+class _CupertinoRefreshControlDemoState
+    extends State<CupertinoRefreshControlDemo> {
+  static const listCount = 20;
+  List<int> randomList;
+
+  @override
+  void initState() {
+    super.initState();
+    _repopulateList();
+  }
+
+  void _repopulateList() {
+    final random = Random();
+    randomList = List<int>.generate(listCount, (i) => i + 1)..shuffle(random);
```

Comment:
nit: the shuffling is the repopulating part, the actual list generation only needs to happen once. 

So this line would change to `randomList.shuffle(random)` and above, you'd initialize it with: 

```dart
final randomList = List<int>.generate(listCount, (i) => i + 1);
```

Reply:
Done